packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx

```diff
@@ -33,31 +33,36 @@ function DefaultChartsAxisTooltipContent(props: ChartsAxisContentProps) {
         )}
 
         <tbody>
-          {series.map(({ color, id, label, valueFormatter, data }: ChartSeriesDefaultized<any>) => {
-            const formattedValue = valueFormatter(data[dataIndex]);
-            if (formattedValue == null) {
-              return null;
-            }
-            return (
-              <ChartsTooltipRow key={id} className={classes.row}>
-                <ChartsTooltipCell className={clsx(classes.markCell, classes.cell)}>
-                  <ChartsTooltipMark
-                    ownerState={{ color }}
-                    boxShadow={1}
-                    className={classes.mark}
-                  />
-                </ChartsTooltipCell>
+          {series
+            .filter((item): item is ChartSeriesDefaultized<CartesianChartSeriesType> =>
+              ['bar', 'line', 'scatter'].includes(item.type),
+            )
```

Comment:
Nitpick: WDYT, would it make sense to extract it into a variable outside of JSX? 🤔

Reply:
Very good idea. I also did it for the container